tests/integration/awslambda/test_lambda.py

```diff
@@ -696,6 +696,22 @@ def test_url_config_lifecycle(self, lambda_client, create_lambda_function, snaps
             lambda_client.get_function_url_config(FunctionName=function_name)
         snapshot.match("failed_getter", ex.value.response)
 
+    def test_large_environment_variables(self, lambda_client, create_lambda_function, snapshot):
+        """Lambda functions with environment variables larger than 4 KiB should fail to create."""
+        snapshot.add_transformer(snapshot.transform.lambda_api())
+        large_envar = "x" * 5 * 1024
```

Comment:
It would be nice if we could pinpoint the size exactly. like test for 4096 bytes and 4097 bytes (or 4095 and 4096, if the limit is exclusive).